joshuaCarpentier/assignments/cyclicSort/cyclicSortEasy/cyclicSortEasy.js

```diff
@@ -1,11 +1,48 @@
 // Problem Statement #
 
-// We are given an array containing ‘n’ objects. Each object, when created, was assigned a unique number from 1 to ‘n’ based on their creation sequence. This means that the object with sequence number ‘3’ was created just before the object with sequence number ‘4’.
+// We are given an array containing ‘n’ objects. Each object, when created, was assigned a unique number from 1 to ‘n’ 
+// based on their creation sequence. This means that the object 
+// with sequence number ‘3’ was created just before the object with sequence number ‘4’.
 
-// Write a function to sort the objects in-place on their creation sequence number in O(n) and without any extra space. For simplicity, let’s assume we are passed an integer array containing only the sequence numbers, though each number is actually an object.
+// Write a function to sort the objects in-place on their creation sequence number in O(n) and
+//  without any extra space. For simplicity, let’s assume we are passed an integer array containing 
+//  only the sequence numbers, though each number is actually an object.
+
+
+/**
+ * Input: Unorder array
+ * Output: Sorted array
+ * 
+ * while current index less then array length if 
+ * current index not equal to index
+ * 
+ * next index  equals current index - 1
+ * 
+ * then inddex  equals next index
+ * 
+ * else iterate over array and return sorted array
+ */
+
+// using pointers sort for when you no were things are espicially for a range of numbers
 
 const cyclic_sort = function (nums) {
-  // TODO: Write your code here
+let index = 0;
+
+while (index < nums.length){
+  let y = nums[index] -1; 
+  if (y != index){
+    let nextIndex = nums[index] - 1;
+    let x = nums[index];
+
+    nums[index] = nums[nextIndex];
+    nums[nextIndex] = x;
+  } else{
+    index++;
+  }
+
+}
+ 
+
```

Comment:
having letters for variable names is extremely popular in competitive programming but might not be as popular for interviewing. I would suggest maybe trying descriptive naming